### DISCUSSION TITLE
Fully qualified name provider is broken

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#738] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.13.0

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I have a Konsist test that checks for an annotation on a class:

```
describe("Final BasePacket subclasses have @Packet annotation") {
    withData(
        nameFn = {
            classNameRegex.find(it.fullyQualifiedName)?.value?.substring(1) ?: it.name
        },
        finalClasses.withParent { abstractNames.contains(it.name) }
    ) { packetClass ->
        listOf(packetClass).assertTrue {
            it.hasAnnotationOf(Packet::class)
        }
    }
}
```

`hasAnnotationOf` uses `fullyQualifiedName` to check that the annotation is of the correct type. The fully qualified name provider searches for that annotation in the imports of the containing file and takes the first import it finds to check the qualified name. If the annotation is not the first import, another import could satisfy the conditions first.

My current workaround is to use `hasAnnotationWithName` instead.

### Actual behaviour

The `fullyQualifiedName` of an annotation named `Packet` was:

```
com.example.PacketReader
```

### Expected behaviour

What it should have been:

```
com.example.protocol.Packet
```

**Comment by nataliapeterwas** [comment, marked as answer]
Thanks for reporting.
This will be fixed in the next release.

**Reply by igorwojda** [reply]
Fixed in https://github.com/LemonAppDev/konsist/pull/830

**Comment by nataliapeterwas** [comment]
A new version v0.14.0 has been released. It contains the fix.
See [v0.14.0 release](https://github.com/LemonAppDev/konsist/releases/tag/v0.14.0)
